godef.go

```diff
@@ -36,20 +37,45 @@ func fail(s string, a ...interface{}) {
 }
 
 func init() {
-	// take GOPATH, set types.GoPath to it if it's not empty.
-	p := os.Getenv("GOPATH")
-	if p == "" {
-		return
+	if os.Getenv("GO15VENDOREXPERIMENT") != "" {
+		setupVendorExperiment()
 	}
-	gopath := strings.Split(p, ":")
-	for i, d := range gopath {
-		gopath[i] = filepath.Join(d, "src")
+}
+
+func setupVendorExperiment() {
+	gopath := []string{}
+
+	flag.Parse()
+
+	dirname := path.Dir(*fflag)
```

Comment:
This doesn't seem right to me. It assumes that we can make a single GOPATH based on the location of the original source file, but that's not how GOVENDOREXPERIMENT works.

For example consider the following packages. If we find the definition of Z in the println line, we should see the field from github.com/x/vendor/b not the method from github.com/m/m/vendor/b, but this PR will print the latter AFAICS.

    // in github.com/m/m
    package main
    import "github.com/x/a"
    func main() {
         var x a.X
         println(x.Y.Z)
    }

    // in github.com/m/m/vendor/b
    package b
    type Y struct{}
    func (Y) Z() {
    }

    // in github.com/x/a
    package a
    import "b"
    type X struct {
         Y b.Y
    }

    // in github.com/x/vendor/b
    package b
    type Y struct {
         Z string
    }